### ISSUE TITLE
Add "mousehide" setting to hide the mouse when typing

### ISSUE THREAD

**Issue by Jelmerro** [Jelmerro/Vieb#45]
Vieb should have Vim's mousehide setting/toggle, to hide the mouse cursor when typing, and show it again when the mouse is moved. These are the requirements for such a feature:

- Always show the mouse cursor when "mousehide" is disabled
- Hide the mouse cursor when the user starts typing with "mousehide" enabled (regardless of other mouse-related settings)
- Show the mouse cursor again when the mouse is moved
- Mousehide should be on by default
- Also have a listeners while in insert mode to implement the same functionality as above

The main reason for the lack of support for this feature is because there is a long-standing Electron bug preventing the cursor from being updated in a webview. I have also tried a couple of workarounds to fix this, but most of them ironically require the mouse to be moved before the cursor styling is applied. See [this electron issue](https://github.com/electron/electron/issues/7747) for details. I will not merge PRs that do not contain a fix/workaround for this bug, as it defeats the purpose of having a "mousehide" setting.

**Comment by Jelmerro** [comment]
Just for reference, I briefly tried to implement this again, it's still blocked by the same upstream issue.

**Comment by FibreFoX** [comment]
Just wanted to drop this here, due to this being referenced an that electron issue. That issue got closed without a proper fix (due to a stupid auto-close bot). You might want to open a new issue on this issue on the electron project. I was having the same issue, but am not working on that project anymore, so I can not contribute with a working reproducable example.

**Comment by Jelmerro** [comment]
Thanks for the heads up, I'll make a new issue in the Electron repo soon :+1:
